integr8ly/integreatly-operator
# MGDAPI-3636 Extended multitenant e2e test to automate RHOAM DevSandbox QuickStart

# Issue link
JIRA: https://issues.redhat.com/browse/MGDAPI-3636

# What
This PR extends the e2e/multitenant-rhoam test to automate the RHOAM DeveloperSandbox [QuickStart guide](https://github.com/briangallagher/sandbox-sre/blob/master/resources/rhoam-ds-prod.xe9u.p1.openshiftapps.com/configure/member/consolequickstart/rhoam-devsandbox-managed-api-quickstart.yaml).

# Verification steps
Passing the PROW tests should be sufficient for verification. However, the test can also be run manually:
1. Provision a cluster
2. Checkout this PR and install RHOAM onto the cluster (either using OLM or locally, it doesn't matter)
3. Once the installation is complete, run the test with `TENANTS_NUMBER=2`:
`TENANTS_NUMBER=2 INSTALLATION_TYPE=multitenant-managed-api BYPASS_STORAGE_TYPE_CHECK=true TEST="M01" make test/e2e/single`
4. Confirm that the test runs to completion, it should take around 5 minutes to run on a fresh cluster
5. Once the test has finished, run it again but this time with `TENANTS_NUMBER=1`:
`TENANTS_NUMBER=1 INSTALLATION_TYPE=multitenant-managed-api BYPASS_STORAGE_TYPE_CHECK=true TEST="M01" make test/e2e/single`
6. Confirm that the test runs to completion, it should take about 3 minutes to run
